agent/functional_tests/tests/functionaltests_unix_test.go

```diff
@@ -618,3 +618,197 @@ func TestNetworkModeBridge(t *testing.T) {
+	maxAttributes := 10
+	customAttributes := `{
+                "key1": "val1",
+                "key2": "val2",
+                "key3": "val3",
+                "key4": "val4",
+                "key5": "val5",
+                "key6": "val6",
+                "key7": "val7",
+                "key8": "val8",
+                "key9": "val9",
+                "key0": "val0"
+        }`
+	os.Setenv("ECS_INSTANCE_ATTRIBUTES", customAttributes)
+
+	agent := RunAgent(t, nil)
+	defer agent.Cleanup()
+
+	params := &ecsapi.DescribeContainerInstancesInput{
+		Cluster:            &agent.Cluster,
+		ContainerInstances: []*string{&agent.ContainerInstanceArn},
+	}
+
+	resp, err := ECS.DescribeContainerInstances(params)
+	require.NoError(t, err)
+	require.NotEmpty(t, resp.ContainerInstances)
+
+	attribMap := AttributesToMap(resp.ContainerInstances[0].Attributes)
+
+	for i := 0; i < maxAttributes; i++ {
+		k := "key" + strconv.Itoa(i)
+		v := "val" + strconv.Itoa(i)
+		assert.Equal(t, v, attribMap[k], "Values should match")
+	}
+}
+
+// TestCustomAttributesWithMaxlenKey tests the ECS_INSTANCE_ATTRIBUTES
+// upon agent registration with a key of maximum supported length
+func TestCustomAttributesWithMaxlenKey(t *testing.T) {
+	maxLen := 128
+	key, value := GetRandomString(maxLen), "dummy"
+
+	customAttribute := fmt.Sprintf("{\"%s\": \"%s\"}", key, value)
+	os.Setenv("ECS_INSTANCE_ATTRIBUTES", customAttribute)
+
+	agent := RunAgent(t, nil)
+	defer agent.Cleanup()
+
+	params := &ecsapi.DescribeContainerInstancesInput{
+		Cluster:            &agent.Cluster,
+		ContainerInstances: []*string{&agent.ContainerInstanceArn},
+	}
+
+	resp, err := ECS.DescribeContainerInstances(params)
+	require.NoError(t, err)
+	require.NotEmpty(t, resp.ContainerInstances)
+
+	attribMap := AttributesToMap(resp.ContainerInstances[0].Attributes)
+	assert.Equal(t, value, attribMap[key], "Values should match")
+}
+
+// TestCustomAttributesWithMaxlenValue tests the ECS_INSTANCE_ATTRIBUTES
+// upon agent registration with a value of maximum supported length
+func TestCustomAttributesWithMaxlenValue(t *testing.T) {
+	maxLen := 128
+	key, value := "dummy", GetRandomString(maxLen)
+
+	customAttribute := fmt.Sprintf("{\"%s\": \"%s\"}", key, value)
+	os.Setenv("ECS_INSTANCE_ATTRIBUTES", customAttribute)
+
+	agent := RunAgent(t, nil)
+	defer agent.Cleanup()
+
+	params := &ecsapi.DescribeContainerInstancesInput{
+		Cluster:            &agent.Cluster,
+		ContainerInstances: []*string{&agent.ContainerInstanceArn},
+	}
+
+	resp, err := ECS.DescribeContainerInstances(params)
+	require.NoError(t, err)
+	require.NotEmpty(t, resp.ContainerInstances)
+
+	attribMap := AttributesToMap(resp.ContainerInstances[0].Attributes)
+	assert.Equal(t, value, attribMap[key], "Values should match")
+}
+
+// TestCustomAttributesWithMaxlen tests the ECS_INSTANCE_ATTRIBUTES
+// upon agent registration with a key, value of maximum supported length
+func TestCustomAttributesWithMaxlen(t *testing.T) {
```

Comment:
This is more of a unit/integ test for ECS; I'm not sure if it warrants a functional test in the Agent.